vendor/engines/sanger_sequencing/app/models/sanger_sequencing/sample.rb

```diff
@@ -4,6 +4,20 @@ class Sample < ActiveRecord::Base
 
     self.table_name = "sanger_sequencing_samples"
     belongs_to :submission
+    # customer_sample_id is based off of the ID, which we don't have until
+    # after the initial persistence.
+    after_create { customer_sample_id && save }
+
+    def customer_sample_id
+      self[:customer_sample_id] ||= default_customer_sample_id
+    end
+
+    private
+
+    def default_customer_sample_id
+      # last four digits of the id as a zero-padded string
+      id.to_s.last(4).rjust(4, "0")
```

Comment:
I don't know if it's much more clear but I would have gone with `format("%04d", id).last(4)`, presuming `id` won't never be `nil` here.